programs/witness_node/main.cpp

```diff
@@ -155,6 +157,10 @@ int main(int argc, char** argv) {
    {
       elog("Exiting with error:\n${e}", ("e", unhandled_exception->to_detail_string()));
 
+      chain::block_database block_id_to_block;
+      block_id_to_block.open(data_dir / "database" / "block_num_to_block");
```

Comment:
We should not put logic about underlying files in `main()`. Please do it in database class.

Reply:
OK. Thanks !